salt/sdb/yaml.py

```diff
@@ -52,7 +57,14 @@
 }
 
 
-def set_(*args, **kwargs):
+def __virtual__():
+    # Pack magic dunders into GPG renderer
+    setattr(salt.renderers.gpg, '__salt__', {})
```

Comment:
IMHO this will be very brittle, especially if the requirements for the gpg renderer changes and it starts calling other functions that typically exist inside the `__salt__` dunder. IMHO, we should instead pack all of `__salt__` into the `sdb` namespace using the salt loader.

Reply:
Thank you for the feedback, @cachedout. I have never found any information on how to "cross-call" pluggable modules. Here is another example of where I'm calling one pillar from another, using a different technique to deal with the dunders: https://github.com/amendlik/gitstack-pillar. I welcome any advice on the best way to handle this.

Here is what I was thinking on the empty dictionary for `__salt__`:
1. SDB modules do not have the `__salt__` dunder in the namespace. I suppose that could be changed, but that's how it works now.
2. The GPG renderer doesn't strictly require it. It falls back to `__opts__` (which is available) if `__salt__` is not there.
3. This feature will probably only be used for master-side SDB operations. It requires the private key to do decryption, and people should not be distributing their private key to minions. I believe this limits any potential dependency on `__salt__` and the functions it provides.

Reply:
IMHO, we should introduce the `__salt__` dunder into sdb. To do so wouldn't be too hard. We would need to create a copy of those functions (or pass them in where possible) to `salt.utils.sdb` and then modify `salt.loader.sdb()` to accept those functions. Finally, pack them into the sdb functions by using the `pack` arg. Let me know if I can explain that a little better. :]